src/symbolic.ml

```diff
@@ -6,7 +6,29 @@ open Types
 module Def_value = Concrete_value
 module Solver = Thread.Solver
 
-module P = struct
+module type Thread = sig
+  type t
+
+  type 'a solver_module = (module Encoding.Solver_intf.S with type t = 'a)
+  type solver = S : 'a solver_module * 'a -> solver
```

Comment:
This and

```ocaml
  val solver: t -> solver

  val pc: t -> Symbolic_value.S.vbool list
```

were added unsupervised by me and @epatrizio so you may want to use double the amount of caution when reviewing.

Reply:
I'm not sure this is what we want, but you'll find out while writing the fuzzer part. :-)